fili-core/src/main/java/com/yahoo/bard/webservice/data/DruidHavingBuilder.java

```diff
@@ -96,7 +130,17 @@ public static Having buildHaving(LogicalMetric metric, ApiHaving having) {
         //     !(metric1 > 2 || metric1 > 4 || metric1 > 8)
         // . In both cases, the second method is correct.
 
-        Having newHaving = havings.size() == 1 ? havings.get(0) : new OrHaving(havings);
-        return operation.isNegated() ? new NotHaving(newHaving) : newHaving;
+        Having newHaving;
+        if (operation.equals(HavingOperation.between)) {
```

Comment:
You can dramatically reduce the cyclomatic complexity of this method by restructuring it a little.  Use the same variable (say 'newHaving') in both of your initial if/else.  (Moving the if count == 1 inside the else.
Then you can just return on the final line with the not-transform not even needing to be assigned to a local variable.